include/i2c.h

```diff
@@ -175,6 +232,116 @@ static inline int _impl_i2c_transfer(struct device *dev,
 	return api->transfer(dev, msgs, num_msgs, addr);
 }
 
+/**
+ * @brief Registers the provided config as Slave device
+ *
+ * Enable I2C slave mode for the 'dev' I2C bus driver using the provided
+ * 'config' struct containing the functions and parameters to send bus
+ * events. The I2C slave will be registered at the address provided as 'address'
+ * struct member. Addressing mode - 7 or 10 bit - depends on the 'flags'
+ * struct member. Any I2C bus events related to the slave mode will be passed
+ * onto I2C slave device driver via a set of callback functions provided in
+ * the 'callbacks' struct member.
+ *
+ * Most of the existing hardware allows simultaneous support for master
+ * and slave mode. This is however not guaranteed.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param cfg Config struct with functions and parameters used by the I2C driver
+ * to send bus events
+ *
+ * @retval 0 Is successful
+ * @retval -EINVAL If parameters are invalid
+ * @retval -EIO General input / output error.
+ * @retval -ENOTSUP If slave mode is not supported
+ */
+__syscall int i2c_slave_register(struct device *dev,
```

Comment:
Where is the Z_SYSCALL_HANDLER for this one? Can't find it in the PR.

Reply:
Same here